### PR TITLE
port marshalling code to napi

### PR DESCRIPTION
**Pull request by paulfitz**
This brings our fork up to date with a version of upstream just prior to dropping support for node v10.  Ports our modifications to use NAPI.  With this change, Grist can be built against node v12.  All tests pass, including marshalling, and excluding electron. The only changes here that are not simple merges are in this commit: https://github.com/gristlabs/node-sqlite3/commit/c2dc8a9f26b6a2139f037a201d321ca650940201

### CODE REVIEW

File: src/database.cc [hunk 1]
```diff
@@ -235,8 +236,11 @@ void Database::Work_BeginClose(Baton* baton) {
     baton->db->RemoveCallbacks();
     baton->db->closing = true;
 
-    int status = uv_queue_work(uv_default_loop(),
-        &baton->request, Work_Close, (uv_after_work_cb)Work_AfterClose);
+    Napi::Env env = baton->db->Env();
```
Comment (dsagal) [inline src/database.cc:239]: Is this bit from merging? Doesn't seem related to marshalling...

**Review by dsagal — APPROVED** [review]

File: src/database.cc (hunk 1 above)
Comment (paulfitz) [reply src/database.cc:239]: Yes, I'm not sure why that happened.
